mw_jwt_test.go

```diff
@@ -804,8 +909,36 @@ func BenchmarkJWTSessionRSAWithEncodedJWK(b *testing.B) {
 			b,
 			test.TestCase{
 				Headers: authHeaders,
-				Code:    200,
+				Code:    http.StatusOK,
 			},
 		)
 	}
 }
+
+func TestJWTHMACIdNewClaim(t *testing.T) {
+	ts := newTykTestServer()
+	defer ts.Close()
+
+	//If we skip the check then the Id will be taken from SUB and the call will succeed
+	_, jwtToken := prepareGenericJWTSession(t.Name(), HMACSign, "user-id", false, true)
+	authHeaders := map[string]string{"authorization": jwtToken}
+	t.Run("Request with valid JWT/HMAC signature/id in user-id claim", func(t *testing.T) {
+		ts.Run(t, test.TestCase{
+			Headers: authHeaders, Code: http.StatusOK,
+		})
+	})
+}
+
+func TestJWTRSAIdNewClaim(t *testing.T) {
+	ts := newTykTestServer()
+	defer ts.Close()
+
+	//If we skip the check then the Id will be taken from SUB and the call will succeed
+	_, jwtToken := prepareGenericJWTSession(t.Name(), RSASign, "user-id", false, true)
+	authHeaders := map[string]string{"authorization": jwtToken}
+	t.Run("Request with valid JWT/RSA signature/id in user-id claim", func(t *testing.T) {
+		ts.Run(t, test.TestCase{
+			Headers: authHeaders, Code: http.StatusOK,
+		})
+	})
+}
```

Comment:
case when `getUserIdFromClaim` if falling back to SUB is not covered by tests

Reply:
Added TestJWTRSAIdInClaimsWithBaseField, TestJWTRSAIdInClaimsWithoutBaseField.